### DISCUSSION TITLE
Is there a way to create BOM without knowing the tech used?

### DISCUSSION THREAD

**Discussion by kartik-579** [DependencyTrack/dependency-track#1551] [Q&A]
Hi,

I have a use case where DT is to be used as a plugin and to be run pre/post CI build stage (generate, upload BOM to a new project and get that project findings). I have a couple of questions related to the same - 

1. Is there a way to create BOM from a git repository without knowing the language the project is written in ? 
2. I was not able to find UUID of a newly created project in the response of api used for uploading BOM. Is there a way to get this UUID?

Also, I looked into the jenkins plugin for DT and did not found any input for getting the tech used in project. This is in some way the same usecase as mine and if it might help...Where can i look for the plugin's implementation ?

Thanks

**Comment by ruckc** [comment]
[cdxgen](https://github.com/AppThreat/cdxgen) is the closest tool i've found, but working with gradle can sometimes be iffy as sometimes you need Java 8, sometimes you need a newer version.

**Reply by kartik-579** [reply]
Thanks for the quick answer, will surely look into the tool.
